src/rospack.cpp

```diff
@@ -331,16 +331,13 @@ Rosstackage::crawl(std::vector<std::string> search_path,
 {
   if(!force)
   {
-    if(readCache())
+    // read the cache only once (when search_paths_ is still empty)
+    if(search_paths_.empty() && readCache())
```

Comment:
Why do you want to prevent using the existing cache here? This will result in worse performance when this function is called multiple times.